Program.cs

```diff
@@ -0,0 +1,545 @@
+    /// </remarks>
+    internal static class Program
+    {
+
+        /// <summary>
+        /// Main entry point for NTRightsNet application.
+        /// </summary>
+        public static int Main([CanBeNull, ItemCanBeNull] string[] args)
+        {
+            // determine command to run
+            var sanitizedArgs = args?.Select(a => a?.Trim() ?? string.Empty).ToArray() ?? new string[0];
+            var commandSpec = sanitizedArgs.Length < 1 ? string.Empty : sanitizedArgs[0];
+            if (!Program.Command.TryGetValue(commandSpec, out var command)) { command = Program.CommandNotFound; }
+
+            // run command and display its result
+            var result = command(args);
+            switch (result)
+            {
+                case Success<string> s:
+                    s.Value.WriteToConsoleWithColour();
+                    Console.WriteLine();
+                    return 0;
+
+                case Failure<string> e:
+                    $"&darkRed;Error: &red;{e.Message}\n".WriteToConsoleWithColour();
+                    return 1;
+
+                default:
+                    "&darkRed;Unknown error\n".WriteToConsoleWithColour();
+                    return 2;
+            }
+        }
+
+
+
+        //**************************************************
+        //* Private
+        //**************************************************
+
+        //--------------------------------------------------
+        /// <summary>
+        /// Defines recognized commands.
+        /// </summary>
+        private static readonly Dictionary<string, Func<string[], Result<string>>> Command = new Dictionary<string, Func<string[], Result<string>>>(StringComparer.OrdinalIgnoreCase)
+        {
+            { string.Empty, Program.CommandUsage },
+            { "help", Program.CommandUsage },
+            { "--help", Program.CommandUsage },
+            { "-h", Program.CommandUsage },
+            { "/h", Program.CommandUsage },
+            { "-?", Program.CommandUsage },
+            { "/?", Program.CommandUsage },
+
+            { "version", Program.CommandGetVersion },
+            { "--version", Program.CommandGetVersion },
+            { "-v", Program.CommandGetVersion },
+
+            { "get", Program.CommandGetRights },
+
+            { "add", Program.CommandAddRights },
+
+            { "remove", Program.CommandRemoveRights }
+        };
+
+
+        //--------------------------------------------------
+        /// <summary>
+        /// Adds rights for the specified user.
+        /// </summary>
+        [NotNull] private static Result<string> CommandAddRights([NotNull, ItemNotNull] string[] args)
+        {
+            // verify arguments
+            if (args.Length != 3)
+            {
+                return new Failure<string>($"Expected exactly 2 parameters, username and semicolonSeparatedRights\n\n{((Success<string>)Program.CommandUsage(args)).Value}");
```

Comment:
`((Success<string>)Program.CommandUsage(args)).Value` is used four times.  How about defining a constant `Usage` that `CommandUsage` returns so you can use it for these cases?

It would be nice if it was declared right next to the command dictionary to make keeping them in sync easier and more obvious.

Reply:
Good idea, updated.